### PR TITLE
Add ability to convert ClickHouse table structure to CapnProto/Protobuf format schema

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add new functions `structureToCapnProtoSchema`/`structureToProtobufSchema` that convert ClickHouse table structure to CapnProto/Protobuf format schema. Allow to intput/output data in CapnProto/Protobuf format without external format schema using autogenerated schema from table structure (controled by settings `format_capn_proto_use_autogenerated_schema`/`format_protobuf_use_autogenerated_schema`). Allow to export autogenerated schema while input/outoput using setting `output_format_schema`.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit b300781fd8995a4e8feac0e58a9b756c17fe93f7 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52278/b300781fd8995a4e8feac0e58a9b756c17fe93f7/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

File: src/Formats/StructureToFormatSchemaUtils.cpp
Comment (antonio2368) [inline src/Formats/StructureToFormatSchemaUtils.cpp:64]:
```suggestion
        return {std::move(first), std::move(second)};
```

File: src/Formats/StructureToProtobufSchema.cpp
Comment (antonio2368) [inline src/Formats/StructureToProtobufSchema.cpp:42]: why not `int32` and `int64` like in capnproto?

Comment (Avogar) [reply src/Formats/StructureToProtobufSchema.cpp:42]: Because we serialize/deserialize Decimal32/64 in a different way. In CapnProto format we serialize Decimal32/64 as their underlying int32/64 values into CapnProto int32/64. In Protobuf if we have int32/64 for Decimal32/64 in the schema, we will cast decimal values to integer values, so we will loose precision. So, I decided that better to use `bytes`  by default for Protobuf format to not loose precision

**Comment by antonio2368** [comment]
LGTM, after you resolve the comment for loop I'll approve (want to avoid auto merge)

File: src/Formats/StructureToFormatSchemaUtils.cpp
Comment (Avogar) [reply src/Formats/StructureToFormatSchemaUtils.cpp]: Done. Also added a comment why we need first lower-case letters

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**
